Core/include/Acts/Geometry/DetectorVolume.hpp

```diff
@@ -346,6 +357,9 @@ class DetectorVolume : public std::enable_shared_from_this<DetectorVolume> {
   /// Volume store (internal/external)
   ObjectStore<std::shared_ptr<DetectorVolume>> m_volumes;
 
+  /// BoundingBox
+  std::shared_ptr<const BoundingBox> m_boundingBox;
```

Comment:
Just wondering why this is a `shared_ptr`, given it is returned as a reference in `getBoundingBox`? Obviously it's not a huge overhead, so this is mostly curiosity.

cc @paulgessinger

Reply:
It needs to be owned somewhere, and we try to avoid owning by raw pointer. I suppose this could be a member variable if we're sure it will always have one.

Reply:
Should I make a change or am I good to mark this as resolved?

Reply:
@paulgessinger I definitely meant member variable.

If we don't make this change, what will be the behaviour of `getBoundaryBox()`?
```cpp
const Acts::Experimental::DetectorVolume::BoundingBox&
Acts::Experimental::DetectorVolume::getBoundingBox() const {
  return *m_boundingBox;
}
```
Isn't this returning a reference to a null pointer? 

At least, when I tried this locally:
```cpp
struct Base
{
    unsigned int a;
}

    std::shared_ptr<Base> p;
    Base& test = *p;
    std::cout<<"p.get() "<<p.get()<<std::endl;
```

I get:
```
p.get() 0
```

Reply:
If the `shared_ptr` is empty you get a `nullptr` for sure. I suppose the idea with `shared_ptr` would be to ensure it's non-null by calling the constructor function from all the constructors. 

I think I agree that just making it a member variable and making sure to initialize it would probably be preferable.

I believe you can do something like

```cpp
class DetectorVolume {
  static BoundingBox createBoundingBox(Args...);
  BoundingBox m_bb;

  DetectorVolum() : m_bb{createBoundingBox(...)} {}
};
```

if you want to centralize the code to construct the BB. I believe here the function would have to be static, i.e. you'd have to pass in the required info by hand, because `this` is not available yet (double check me on that).

Reply:
Hi @paulgessinger, maybe I'm just not understanding but I'm not entirely sure I can build a BB at construction because the portals aren't yet defined which are needed in making the BB.

Reply:
We discussed this, and I think @Matthewharri is right here.